### PR TITLE
Document functions, add features: 'return to menu', 'remove tiles'

### PR DESCRIPTION
**Pull request by BurhanHTW**
- Implemented 'Return to Menu' functionality.
- Added 'Remove Tiles' feature.
- Unified saving game state and statistics into a single file.
- Enabled selection of saved game states for continuation.

### CODE REVIEW

**Comment by BurhanHTW** [comment]
We have submitted the pull request, and it has been a pleasure to contribute to this project.

File: src/.vscode/c_cpp_properties.json
Comment (plibither8) [inline src/.vscode/c_cpp_properties.json:1]: Let's remove the `.vscode` files. They're meant for developer-specific configs on local machines, so shouldn't be part of the repository.

Bonus: Add the `.vscode` folder to the `.gitignore` file. This should prevent it from being staged and committed in the future as well! :)

File: src/.vscode/launch.json
Comment (plibither8) [inline src/.vscode/launch.json:1]: Same with this file... let's remove it.

File: src/.vscode/settings.json
Comment (plibither8) [inline src/.vscode/settings.json:1]: ...and this one too

File: src/game-graphics.cpp
Comment (plibither8) [inline src/game-graphics.cpp:130]: Comments look great, thank you!

File: src/game-graphics.cpp
Comment (plibither8) [inline src/game-graphics.cpp:127]: We can remove this note, since it will get outdated pretty quickly.

File: src/game-pregamemenu.cpp
Comment (plibither8) [inline src/game-pregamemenu.cpp:142]: Let's be careful to remove commented out code

File: src/game.cpp
Comment (plibither8) [inline src/game.cpp:32]: Pedantic, but good for uniformity: let's remove the multiple spaces before the comment here. Best to move the comment above `FLAG_TILES_REMOVED`.

File: src/game.cpp
Comment (plibither8) [inline src/game.cpp:224]: I see commented out code 👀

File: src/game.cpp
Comment (plibither8) [inline src/game.cpp:330]: Same with the note here! My suggestion: move these notes to the PR description.

File: src/gameboard.cpp
Comment (plibither8) [inline src/gameboard.cpp:152]: commented out code 👀

File: src/loadresource.cpp
Comment (plibither8) [inline src/loadresource.cpp:40]: Great idea here, as well as the `-1` to return an error!

File: src/loadresource.cpp
Comment (plibither8) [inline src/loadresource.cpp:131]: Why was the `'.'` required?

File: src/menu.cpp
Comment (plibither8) [inline src/menu.cpp:248]: Let's move these comments to the line above.

File: src/scores-graphics.cpp
Comment (plibither8) [inline src/scores-graphics.cpp:59]: commented out code 👀

**Review by plibither8 — APPROVED** [review]
Code looks good, thank you very much! :)

I can see that a [CI workflow is failing](https://github.com/plibither8/2048.cpp/actions/runs/9585418900/job/26544757841?pr=145). Could you check the error behind it? Ideally, we want these checks to be successful before trusting it to be merged.

Additionally, a couple of changes I'd love to see before merging it:
* Let's remove the `.vscode` directory. More details in the review comments.
* Let's be uniform with how we do in-line comments. Gives much better readability.
* The added code would benefit from being run against a formater/linter. This will format the code to make it uniform. We've got a [`.clang-format`](https://github.com/plibither8/2048.cpp/blob/master/.clang-format) file that specifies the formatting conventions used in this project.

Nevertheless, I'm approving this PR on the basis of the solid feature code additions -- they look great!

**Comment by BurhanHTW** [comment]
We‘ll try to fix the CI. Hopefully this solved asap :)

**Comment by BurhanHTW** [comment]
I have to switch from C++14 to C++17 in CMakeLists.txt #include <filesystem> in menu.cpp is only supported from C++17 onwards.  https://en.cppreference.com/w/cpp/filesystem

**Comment by plibither8** [comment]
> I have to switch from C++14 to C++17 in CMakeLists.txt #include <filesystem> in menu.cpp is only supported from C++17 onwards.  https://en.cppreference.com/w/cpp/filesystem

Have you tried fixing the issue using the suggestion in the error message? `std::_fs::filesystem`

**Comment by BurhanHTW** [comment]
> > I have to switch from C++14 to C++17 in CMakeLists.txt #include  in menu.cpp is only supported from C++17 onwards.  https://en.cppreference.com/w/cpp/filesystem
> 
> Have you tried fixing the issue using the suggestion in the error message? `std::_fs::filesystem`

Not yet but what will it Change to have an _ ?

**Comment by plibither8** [comment]
@BurhanHTW could you push the changes to your branch to re-trigger the CI?

**Comment by BurhanHTW** [comment]
> @BurhanHTW could you push the changes to your branch to re-trigger the CI?

I will Push it later when i am at home. We have changed everything what you wanted :)

**Comment by BurhanHTW** [comment]
I cant push anything because the access is denied

**Comment by BurhanHTW** [comment]
I had to generate a new Token to push. Now we are waiting for the workflow approval. Hopefully its done :D

**Comment by plibither8** [comment]
@BurhanHTW your latest commit has many merge conflicts present. [Example](https://github.com/plibither8/2048.cpp/pull/145/commits/054d427eaa0541892a501adb8d9199c796e07f5d#diff-9177f1b4af50ab58a32411f440af9c4bb4f99beabe96721c913a0d8e9dac605dR127)

Remove all merge markers - they're causing the builds to fail.

**Comment by BurhanHTW** [comment]
> @BurhanHTW your latest commit has many merge conflicts present. [Example](https://github.com/plibither8/2048.cpp/pull/145/commits/054d427eaa0541892a501adb8d9199c796e07f5d#diff-9177f1b4af50ab58a32411f440af9c4bb4f99beabe96721c913a0d8e9dac605dR127)
> 
> Remove all merge markers - they're causing the builds to fail.

hopefully done :D

**Comment by BurhanHTW** [comment]
I don’t know what i have Done actually i have to Check the Problems …

**Comment by BurhanHTW** [comment]
As you can see in the commits i have deleted all markers. Maybe it will run without Problems now.

**Comment by BurhanHTW** [comment]
If i have to many Problems. Is it possible to make a new pull Request? I guess i had to many Problems because i commited to many files

**Comment by BurhanHTW** [comment]
If the workflow fails again i‘ll create a new Pull Request and Push the latest Code again. I am sorry to have those issues. :/

**Comment by BurhanHTW** [comment]
So the workflow failed again. I will create a new pull Request.

**Comment by plibither8** [comment]
Yep, that sounds like a good idea!

**Comment by BurhanHTW** [comment]
I have closed this pull request. I will create a new one now.

**Pull request closed without merging**
